File: data/readable_format/cpp/file2173.cpp
#include <iostream>
#include <array>

// Declaration of foo
void foo(int n1, std::array<int64_t, 3> n2, int n3, int64_t& n4) {
    n4 = (n1 + n2[2]) / 2 + n3;
}

// Unit test for foo
void test_foo() {
    int n1 = 2, n3 = 1;
    std::array<int64_t, 3> n2 = {1, 2, 3};
    int64_t n4;

    foo(n1, n2, n3, n4);

    if (n4 != 3) {
        std::cerr << "Test failed: Expected 3, got " << n4 << std::endl;
    } else {
        std::cout << "Test passed: n4 = " << n4 << std::endl;
    }
}

int main() {
    test_foo();
    return 0;
}

File: data/readable_format/fortran/file2173.f90
! Main subroutine
subroutine foo(n1,n2,n3,n4)
    integer :: n1, n3
    integer(kind=8) :: n2(3), n4
    n4 = (n1 + n2(3))/2 + n3
end subroutine foo

! Unit test for the subroutine
program test_foo
    implicit none
    integer :: n1, n3
    integer(kind=8) :: n2(3), n4

    ! Test 1
    n1 = 2
    n2 = (/1_8, 2_8, 3_8/)
    n3 = 1
    call foo(n1, n2, n3, n4)
    if (n4 /= 3) then
        print *, "Test failed: Expected 3, got", n4
    else
        print *, "Test 1 passed: n4 = ", n4
    endif

end program test_foo